### PR TITLE
Adding support for channel search and autocomplete.

### PR DESCRIPTION
**Pull request by crspeller**

### CODE REVIEW

**Review by lieut-data — APPROVED** [review]

File: cmd/loadtest/loadtest.go [hunk 1]
```diff
@@ -52,6 +52,11 @@ var tests []TestItem = []TestItem{
 		ShortDesc: "Test leaving and joining a team while under load",
 		Test:      &loadtest.TestLeaveJoinTeam,
 	},
+	{
+		Name:      "autocomplete",
```
Comment (lieut-data) [inline cmd/loadtest/loadtest.go]: Would it make sense to eventually converge on a definition of "test all" that tests all these types of actions with some reasonable distribution between the entities?

Comment (crspeller) [reply cmd/loadtest/loadtest.go]: That's the goal of "test all"

Comment (lieut-data) [reply cmd/loadtest/loadtest.go]: Ah, of course, I hadn't fully dived into the definition of all these structs and how they are used. 👍

**Review by jwilander — APPROVED** [review]

**Review by jwilander — APPROVED** [review]

**Comment by jwilander** [comment]
Just needs a rebase and good to go

**Comment by lieut-data** [comment]
For posterity, here's a loadtest run against 5.0.0 with these changes:

## Loadtest Results
### Score: 239.03
The score is the average of the 95th percentile, median and interquartile ranges in the routes below.

### Routes
#### /channels/CID/members
| Metric | Actual |
| --- | --- |
| Hits | 1901 |
| Error Rate | 0.00% |
| Mean Response Time | 3.41ms |
| Median Response Time | 3.00ms |
| 95th Percentile | 4.00ms |

#### /channels/CID/members/me
| Metric | Actual |
| --- | --- |
| Hits | 1450 |
| Error Rate | 0.00% |
| Mean Response Time | 2.62ms |
| Median Response Time | 2.00ms |
| 95th Percentile | 3.00ms |

#### /channels/CID/posts
| Metric | Actual |
| --- | --- |
| Hits | 1450 |
| Error Rate | 0.00% |
| Mean Response Time | 1.82ms |
| Median Response Time | 1.00ms |
| 95th Percentile | 4.00ms |

#### /channels/CID/stats
| Metric | Actual |
| --- | --- |
| Hits | 1450 |
| Error Rate | 0.00% |
| Mean Response Time | 1.14ms |
| Median Response Time | 1.00ms |
| 95th Percentile | 2.00ms |

#### /channels/members/me/view
| Metric | Actual |
| --- | --- |
| Hits | 1450 |
| Error Rate | 0.00% |
| Mean Response Time | 10.78ms |
| Median Response Time | 10.00ms |
| 95th Percentile | 15.00ms |

#### /files
| Metric | Actual |
| --- | --- |
| Hits | 6 |
| Error Rate | 0.00% |
| Mean Response Time | 1213.67ms |
| Median Response Time | 1239.00ms |
| 95th Percentile | 1602.50ms |

#### /files/PID/thumbnail
| Metric | Actual |
| --- | --- |
| Hits | 149 |
| Error Rate | 0.00% |
| Mean Response Time | 50.56ms |
| Median Response Time | 35.00ms |
| 95th Percentile | 106.50ms |

#### /hooks/incoming
| Metric | Actual |
| --- | --- |
| Hits | 49 |
| Error Rate | 0.00% |
| Mean Response Time | 16.29ms |
| Median Response Time | 16.00ms |
| 95th Percentile | 20.00ms |

#### /posts
| Metric | Actual |
| --- | --- |
| Hits | 215 |
| Error Rate | 0.00% |
| Mean Response Time | 31.12ms |
| Median Response Time | 29.00ms |
| 95th Percentile | 37.50ms |

#### /posts/PID/files/info
| Metric | Actual |
| --- | --- |
| Hits | 174 |
| Error Rate | 0.00% |
| Mean Response Time | 0.63ms |
| Median Response Time | 0.00ms |
| 95th Percentile | 3.00ms |

#### /teams/TID/channels
| Metric | Actual |
| --- | --- |
| Hits | 137 |
| Error Rate | 0.00% |
| Mean Response Time | 4.69ms |
| Median Response Time | 4.00ms |
| 95th Percentile | 11.00ms |

#### /teams/TID/channels/autocomplete
| Metric | Actual |
| --- | --- |
| Hits | 260 |
| Error Rate | 0.00% |
| Mean Response Time | 6.25ms |
| Median Response Time | 6.00ms |
| 95th Percentile | 11.00ms |

#### /teams/TID/channels/search
| Metric | Actual |
| --- | --- |
| Hits | 2216 |
| Error Rate | 0.00% |
| Mean Response Time | 4.93ms |
| Median Response Time | 5.00ms |
| 95th Percentile | 9.00ms |

#### /teams/TID/posts/search
| Metric | Actual |
| --- | --- |
| Hits | 44 |
| Error Rate | 0.00% |
| Mean Response Time | 6.30ms |
| Median Response Time | 5.00ms |
| 95th Percentile | 14.50ms |

#### /users/UID/active
| Metric | Actual |
| --- | --- |
| Hits | 500 |
| Error Rate | 0.00% |
| Mean Response Time | 110.99ms |
| Median Response Time | 114.00ms |
| 95th Percentile | 204.00ms |

#### /users/email/UID
| Metric | Actual |
| --- | --- |
| Hits | 500 |
| Error Rate | 0.00% |
| Mean Response Time | 36.70ms |
| Median Response Time | 31.00ms |
| 95th Percentile | 101.00ms |

#### /users/status/ids
| Metric | Actual |
| --- | --- |
| Hits | 13072 |
| Error Rate | 0.00% |
| Mean Response Time | 1.57ms |
| Median Response Time | 1.00ms |
| 95th Percentile | 2.00ms |

**Pull request merged**